extensions/default/src/MergeDataSource/index.js

```diff
@@ -0,0 +1,195 @@
+import { DicomMetadataStore, IWebApiDataSource } from '@ohif/core';
+import { get } from 'lodash';
+
+/**
+ * Returns deduplicated data by comparing the incomingData with the mergedData based on the mergeKey.
+ * @param {Object} options - The options object.
+ * @param {Array} options.mergedData - The global data array.
+ * @param {Array} options.incomingData - The incoming data array.
+ * @param {string} options.mergeKey - The merge key.
+ * @returns {Array} The deduplicated data array.
+ */
+const getDedupedData = ({ mergedData, incomingData, mergeKey }) => {
+  const keys = mergedData.map(r => get(r, mergeKey));
+  return incomingData.filter(r => !keys.includes(get(r, mergeKey)));
+};
+
+/**
+ * Processes deduplicated data and pushes it to mergedData array.
+ * @param {Object} options - The options object.
+ * @param {string} options.sourceName - The name of the data source.
+ * @param {Array} options.mergedData - The global data array.
+ * @param {Array} options.incomingData - The incoming data array.
+ * @param {string} options.mergeKey - The key to merge the data on.
+ * @param {Function} options.tagFunc - The function to tag the data with.
+ */
+const processDedupedData = ({ sourceName, mergedData, incomingData, mergeKey, tagFunc }) => {
+  let data = [];
+  if (mergeKey) {
+    const dedupedData = getDedupedData({
+      mergedData: mergedData.flat(),
+      incomingData,
+      mergeKey,
+    });
+    data = data.concat(dedupedData);
+  } else {
+    data = data.concat(incomingData);
+  }
+  mergedData.push(tagFunc(data, sourceName));
+};
+
+function createMergeDataSourceApi(mergeConfig, UserAuthenticationService, extensionManager) {
+  const { seriesMerge } = mergeConfig;
+  const { dataSourceNames, defaultDataSourceName } = seriesMerge;
+
+  /**
+   * Calls a specified function on all data sources and merges the results.
+   *
+   * @async
+   * @function
+   * @param {string} path - The path of the function to call.
+   * @param {Object} args - The arguments to pass to the function.
+   * @returns {Promise<Array>} - A promise that resolves to an array of merged data.
+   */
+  const callForAllDataSourcesAsync = async (path, args) => {
+    const mergeMap = {
```

Comment:
The merge map can be made global/static as nothing in it is specific to the function itself.

Reply:
Updated, thanks.